app/controllers/export_controller.rb

```diff
@@ -57,6 +57,28 @@ def csv
     send_data csv_result, filename: "Sara-Alert-#{params[:workflow].capitalize}-#{params[:type].capitalize}-#{DateTime.now}.csv"
   end
 
+  def excel_comprehensive_patients
+    redirect_to(root_url) && return unless current_user.can_export?
+
+    # Grab patients to export
+    patients = current_user.viewable_patients.where(isolation: params[:workflow] == 'isolation')
+
+    # Do nothing if issue with request/permissions
+    redirect_to(root_url) && return if patients.nil?
```

Comment:
Would patients ever be nil? It looks like it is still just an ActiveRecord::Relation (even if it finds nothing).

Would suggest redirecting before line 64 if params[:workflow] is not isolation or exposure, or default to exposure. Then, if you still want to return if no patients, do a return if patients.size.positive? or empty?